src/components/Breadcrumbs/index.js

```diff
@@ -3,25 +3,34 @@ import React from "react"
 import Breadcrumbs from "react-router-dynamic-breadcrumbs"
 import { itemCls, wrapperCls } from "./styles"
 import { cx } from "emotion"
-
-const routesList = {
-  "/login": "Login",
-  "/register": "Register",
-  "/visits/details": null,
-  "/visits/details/:id": "View a single Visit",
-  "/visits/create-new": "New Visit",
-  "/visits/edit": null,
-  "/visits/edit/:id": "Edit a Visit",
-  "/visits": "All Visits",
-  "/patients": "Patients",
-  "/patients/details": null,
-  "/patients/details/:id": "Patient Details",
-  "/patients/create-new": "Create New",
-  "/patients/edit": null,
-  "/patients/edit/:id": "Patient Edit"
-}
+import { useI18n } from "providers/I18n"
 
 function BreadcrumbsComponent() {
+  const { t } = useI18n()
+
+  const routesList = {
```

Comment:
Let's make routesList the result of a useMemo call in order to keep the same reference between rerenders.

Reply:
you suggest declaring a function that will return the routesList and call this function as a useMemo callback?

Reply:
Yes exactly. You can write this function directly inside the useMemo